src/ChessBoard/MovePieces.tsx

```diff
@@ -68,67 +85,102 @@ export class MovePieces extends React.Component<Props, State> {
         let x = mousePositions.x
         let y = mousePositions.y
         //Can't move pieces outside of canvas
-        if (x >= 550) x = 550
+        const line = this.props.canvasSize - this.props.imageSize
+        if (x >= line) x = line
         if (x <= 0) x = 0
         if (y <= 0) y = 0
-        if (y >= 550) y = 550
-        this.setState({ mousePosition: { x: x, y: y } });
+        if (y >= line) y = line
+        this.mousePosition = { x: x, y: y };
     }
 
     onMouseMove = (event: MouseEvent) => {
         const mousePositions = this.getMousePositions(event)
         let x = mousePositions.x
         let y = mousePositions.y
+        const i = this.draggingIndex
         // Update position of the piece to be in the middle of the square
-        for (let xPosition = 0; xPosition < 600; xPosition+=75){
-            for (let yPosition = 0; yPosition < 600; yPosition+=75){
-                if ((this.state.mousePosition.x >= xPosition && this.state.mousePosition.x <= xPosition + 75) &&
-                    (this.state.mousePosition.y >= yPosition && this.state.mousePosition.y <= yPosition + 75)){
-                        x = xPosition + 12.5
-                        y = yPosition + 12.5
+        for (let xPos = 0; xPos < this.props.canvasSize; xPos += this.props.squareSize){
```

Comment:
Ineffective logic in mouse move handler with O(n^2) complicity. You can adjust mouse position in onMouseUp handler. Here you can just update this.mousePosition = mousePositions; 

Code example for adjust function which you could call in onMouseUp handler:

```
adjustMousePosition = () => {
        let nextSquere = this.props.squareSize;
        const pieceImageShift = (this.props.squareSize - this.props.imageSize) / 2; // also use relative units istead of harcoded numbers

        for (let squere = 0; squere !== this.props.squareSize * 8; squere += this.props.squareSize) {
            if (this.mousePosition.x >= squere && this.mousePosition.x <= nextSquere) {
                this.mousePosition.x = squere + pieceImageShift;
            }

            if (this.mousePosition.y < nextSquere && this.mousePosition.y >= squere) {
                this.mousePosition.y = squere + pieceImageShift;
            }
            nextSquere += this.props.squareSize;
        }

        const positions = [...this.currentPositions];
        positions[this.draggingIndex] = { x: this.mousePosition.x, y: this.mousePosition.y };
        this.currentPositions = positions;
    }
```
```
    onMouseMove = (event: MouseEvent) => {
        const mousePositions = this.getMousePositions(event)
        this.mousePosition = mousePositions;
    }
    
    onMouseUp = () => {
        this.adjustMousePosition()
        this.isDragging = false
        this.draggingIndex = -1
    }
```